src/memo/memo.ml

```diff
@@ -890,7 +901,19 @@ end = struct
              due to diamond dependencies. We could add [Fiber.collect_error_set]
              that returns a set of errors instead of a list, to get rid of the
              duplicates as early as possible. *)
-          Error (Exn_set.of_list exns)
+          let exns, reproducible =
+            List.fold_left exns ~init:(Exn_set.empty, true)
+              ~f:(fun (acc, acc_reproducible) exn ->
+                let exn, acc_reproducible =
+                  match exn with
+                  | { Exn_with_backtrace.exn = Non_reproducible exn; backtrace }
```

Comment:
Right now, I'm erasing the `Non_reproducible` wrapper, so the exceptions are reraised without it. Maybe this is actually confusing and I should erase the wrapper?